src/main/java/de/tum/in/www1/artemis/service/TextBlockService.java

```diff
@@ -99,4 +103,31 @@ public void deleteForSubmission(TextSubmission textSubmission) {
         textBlockRepository.deleteAllBySubmission_Id(textSubmission.getId());
     }
 
+    /**
+     * Sets number of potential automatic Feedback's for each block belonging to the `Result`'s submission.
+     * This number determines how many other submissions would be affected if the user were to submit a certain block feedback.
+     * For each TextBlock of the submission, this method finds how many other TextBlocks exist in the same cluster.
+     * This number is represented with the `numberOfAffectedSubmissions` field which is set here for each
+     * TextBlock of this submission
+     *
+     * @param result Result for the Submission acting as a reference for the text submission to be searched.
+     */
+    @Transactional()
```

Comment:
please get rid of Transactional

Reply:
Removed in  [c7ee045](https://github.com/ls1intum/Artemis/pull/3456/commits/c7ee045c55de65b57de8778dc63e4ebde6278884)